Lib/asyncio/locks.py

```diff
@@ -181,18 +181,22 @@ def locked(self):
             self._locked = True
             return True
 
+        _waiters = self._waiters
         fut = self._loop.create_future()
-        self._waiters.append(fut)
+        _waiters.append(fut)
         try:
-            await fut
-            self._locked = True
-            return True
+            try:
```

Comment:
This is very subtle.  Can you add a comment why we use two nested 'try' statements here? I.e. that it's important that 'finally' block is called before 'except CancelledError'

Reply:
Sure thing :-)